messaging/src/main/java/org/axonframework/messaging/timeout/AxonTimeLimitedTask.java

```diff
@@ -148,6 +149,67 @@ public void complete() {
         }
     }
 
+    /**
+     * Even though the task was processed successfully, it might have been interrupted while processing, and the
+     * exception might have been caught and swallowed by a lower component. This happens, for example, by the
+     * {@link org.axonframework.eventhandling.LoggingErrorHandler} , which is the default in event processors.
+     * <p>
+     * This function checks if the task was interrupted, and if so, it throws a {@link AxonTimeoutException} to indicate
+     * that the processing was aborted due to a timeout. If the task was not interrupted, it checks if the thread was
+     * interrupted. If it was, it throws an {@link InterruptedException} to indicate that the processing was aborted due
+     * to an interrupt. This effectively restores the proper error status, so upper components can handle it.
+     */
+    public void ensureNoInterruptionWasSwallowed() throws InterruptedException {
```

Comment:
Nit: I think it might be good to mention that this exact method is a pure necessity to be invoked when using the `AxonTimeLimitedTask`. So, a short paragraph/sample in the class-level JavaDoc about this, would be a nice addition.

Reply:
Good point, adding!